### PR TITLE
Typescript/music

### PR DESCRIPTION
**Pull request by rvilarl**

### CODE REVIEW

File: src/music.ts [hunk 1]
```diff
@@ -168,18 +169,18 @@ export class Music {
...
     return true;
   }
 
-  isValidIntervalValue(interval) {
+  isValidIntervalValue(interval: number): boolean {
     return this.isValidNoteValue(interval);
   }
 
-  getNoteParts(noteString) {
+  getNoteParts(noteString: string): IMusicNoteParts {
```
Comment (wassertim) [inline src/music.ts:183]: let's keep only IMusicNoteParts

File: src/music.ts [hunk 2]
```diff
@@ -110,18 +111,18 @@ export class Music {
...
     };
   }
 
-  static get accidentals() {
+  static get accidentals(): string[] {
     return ['bb', 'b', 'n', '#', '##'];
   }
 
-  static get noteValues() {
+  static get noteValues(): { [value: string]: IMusicNoteValue } {
```
Comment (wassertim) [inline src/music.ts:125]: avoid dynamic type definition

Comment (wassertim) [reply src/music.ts:125]: BTW it is a Record<string, IMusicNoteValue>

File: src/music.ts [hunk 3]
```diff
@@ -110,18 +111,18 @@ export class Music {
     };
   }
 
-  static get scaleTypes() {
+  static get scaleTypes(): { [scaleType: string]: number[] } {
```
Comment (wassertim) [inline src/music.ts:114]: Record

File: src/music.ts [hunk 4]
```diff
@@ -96,12 +97,12 @@ export class Music {
...
       M7: 11,
       maj7: 11,
-      8: 12,
+      '8': 12,
       octave: 12,
     };
   }
 
-  static get scales() {
+  static get scales(): { [scale: string]: number[] } {
```
Comment (wassertim) [inline src/music.ts:105]: Record

File: src/music.ts [hunk 5]
```diff
@@ -74,20 +75,20 @@ export class Music {
       b3: 3,
       min3: 3,
       M3: 4,
-      3: 4,
+      '3': 4,
```
Comment (wassertim) [inline src/music.ts:78]: let's make it consistent

File: src/music.ts [hunk 6]
```diff
@@ -56,7 +57,7 @@ export class Music {
     };
   }
 
-  static get intervals() {
+  static get intervals(): { [interval: string]: number } {
```
Comment (wassertim) [inline src/music.ts:60]: Record

File: src/music.ts [hunk 7]
```diff
@@ -30,15 +31,15 @@ export class Music {
...
     return ['c', 'c#', 'd', 'd#', 'e', 'f', 'f#', 'g', 'g#', 'a', 'a#', 'b'];
   }
 
-  static get diatonic_intervals() {
+  static get diatonic_intervals(): string[] {
     return ['unison', 'm2', 'M2', 'm3', 'M3', 'p4', 'dim5', 'p5', 'm6', 'M6', 'b7', 'M7', 'octave'];
   }
 
-  static get diatonic_accidentals() {
+  static get diatonic_accidentals(): { [diatonic_interval: string]: IMusicAccidental } {
```
Comment (wassertim) [inline src/music.ts:42]: Record

File: src/music.ts [hunk 8]
```diff
@@ -4,21 +4,22 @@
...
     return ['c', 'd', 'e', 'f', 'g', 'a', 'b'];
   }
 
-  static get root_values() {
+  static get root_values(): number[] {
     return [0, 2, 4, 5, 7, 9, 11];
   }
 
-  static get root_indices() {
+  static get root_indices(): { [root: string]: number } {
```
Comment (wassertim) [inline src/music.ts:22]: Record

**Review by wassertim — CHANGES_REQUESTED** [review]

File: src/music.ts (hunk 5 above)
Comment (rvilarl) [reply src/music.ts:78]: Left like in master. Lint makes automatics changes and it is not worthy to get other files changed.

File: src/music.ts [hunk 9]
```diff
@@ -110,18 +111,18 @@ export class Music {
...
 
-  static get accidentals() {
+  static get accidentals(): string[] {
     return ['bb', 'b', 'n', '#', '##'];
   }
 
-  static get noteValues() {
+  static get noteValues(): Record<string, IMusicNoteValue> {
     return {
       c: { root_index: 0, int_val: 0 },
```
Comment (wassertim) [inline src/music.ts:127]: you decided not to touch it?

**Pull request merged**